### PR TITLE
🎨 Use unit library for sidb defects.

### PR DESCRIPTION
**Pull request by Drewniok**
## Description

This PR introduces the use of the ``unit library`` to sidb defects.

## Checklist:

- [x] The pull request only contains commits that are related to it.
- [x] I have added appropriate tests and documentation.
- [ ] I have made sure that all CI jobs on GitHub pass.
- [ ] The pull request introduces no new warnings and follows the project's style guidelines.

### CODE REVIEW

File: test/io/read_sqd_layout.cpp [hunk 1]
```diff
@@ -136,7 +136,7 @@ TEST_CASE("Read single defect SQD layout", "[sqd]")
                                               "          <incl_coords>\n"
                                               "              <latcoord n=\"5\" m=\"2\" l=\"0\" />\n"
                                               "          </incl_coords>\n"
-                                              "          <coulomb charge=\"2\" eps_r=\"-1.2\" lambda_tf=\"3.4\" />\n"
+                                              "          <coulomb charge=\"2\" eps_r=\"1.2\" lambda_tf=\"3.4\" />\n"
```
Comment (marcelwa) [inline test/io/read_sqd_layout.cpp:139]: Why did you change the test here?

Comment (Drewniok) [reply test/io/read_sqd_layout.cpp:139]: eps_r cannot be negative.

Comment (marcelwa) [reply test/io/read_sqd_layout.cpp:139]: Got it, thanks!

**Comment by marcelwa** [comment]
When you fix the Windows CI, I'd gladly approve this PR.

**Review by marcelwa — APPROVED** [review]
